src/commands/testDependenciesCommands.ts

```diff
@@ -0,0 +1,324 @@
+    }
+
+    testProjects = testProjects.filter((project: IJavaTestItem) => {
+        return project.testKind === TestKind.None;
+    });
+
+    if (testProjects.length === 0) {
+        sendError(new Error('Failed to find a project to enable tests.'));
+        return undefined;
+    }
+
+    // currently this feature will only be enabled when workspace contains one unmanaged folder without test dependencies.
+    return testProjects[0];
+}
+
+async function setupUnmanagedFolder(projectUri: Uri): Promise<void> {
+    const testKind: TestKind | undefined = await getTestKind();
+    if (testKind === undefined) {
+        return;
+    }
+    const libFolder: string = await getLibFolder(projectUri);
+    const libFolderExists: boolean = await fse.pathExists(libFolder);
+    if (!libFolderExists) {
+        await fse.ensureDir(libFolder);
+    }
+
+    try {
+        await window.withProgress({
+            location: ProgressLocation.Notification,
+            cancellable: true
+        }, async (progress: Progress<{message?: string; increment?: number}>, token: CancellationToken) => {
+            const metadata: IArtifactMetadata[] = getJarIds(testKind);
+            for (const jar of metadata) {
+                if (token.isCancellationRequested) {
+                    throw new Error('User cancelled');
+                }
+                progress.report({
+                    message: `Downloading ${jar.artifactId}.jar...`,
+                });
+                if (!jar.version) {
+                    jar.version = await getLatestVersion(jar.groupId, jar.artifactId) || jar.defaultVersion;
+                }
+                await downloadJar(libFolder, jar.groupId, jar.artifactId, jar.version, metadata.length, progress, token);
+            }
+        });
+    } catch (e) {
+        if (e?.message !== 'User cancelled') {
+            sendError(e);
+        }
+        if (!libFolderExists) {
+            fse.remove(libFolder);
+        }
+        return;
+    }
+
+    updateProjectSettings(projectUri, libFolder);
+}
+
+async function getTestKind(): Promise<TestKind | undefined> {
+    const framework: any = await window.showQuickPick([{
+        label: 'JUnit Jupiter',
+        value: TestKind.JUnit5,
+    }, {
+        label: 'JUnit',
+        value: TestKind.JUnit,
+    }, {
+        label: 'TestNG',
+        value: TestKind.TestNG,
+    }], {
+        placeHolder: 'Select the test framework to be enabled.'
+    });
+    return framework?.value;
+}
+
+async function getLibFolder(projectUri: Uri): Promise<string> {
+    const referencedLibraries: any = workspace.getConfiguration('java', projectUri).get('project.referencedLibraries');
+    if (_.isArray(referencedLibraries)) {
+        // do a simple check if the project uses default lib location.
+        if (referencedLibraries.includes('lib/**/*.jar')) {
+            return path.join(projectUri.fsPath, 'lib');
+        }
+    }
+
+    for (let i: number = 0; i < 100; i++) {
```

Comment:
what's this `100`?

Reply:
It's used to get a new available folder. For example, if `test-lib` folder already exists, then we will create `test-lib1`

Reply:
It's used to create a new available folder. For example. if `test-lib` already exists, then we will use `test-lib1`

Reply:
ok, I think it works for most cases.